gems/kaminari-core/1.2/kaminari-core.rbs

```diff
@@ -0,0 +1,38 @@
+module Kaminari
+  def self.config: () -> Kaminari::Config
+  def self.configure: () { (Kaminari::Config) -> void } -> void
+
+  class Config
+    public attr_accessor default_per_page: Integer
+    public attr_accessor max_per_page: Integer?
+    public attr_accessor window: Integer
+    public attr_accessor outer_window: Integer
+    public attr_accessor left: Integer
+    public attr_accessor right: Integer
+    public attr_accessor page_method_name: Symbol | String | :page
+    public attr_accessor max_pages: Integer?
+    public attr_accessor params_on_first_page: bool
+
+    public attr_writer param_name: Symbol | String | :page | untyped
+
+    def param_name: () -> (Symbol | String | :page)
+  end
+
+  module PageScopeMethods
+    include Kaminari::_Model_Relation
```

Comment:
`_Model_Relation` provides `limit` and `offset` methods, but actually `PageScopeMethods` doesn't define them. These methods are defined by a module that includes `PageScopeMethods`.


In this case, "self type parameter" feature is useful.
https://github.com/ruby/rbs/blob/6c847d1177e6e03435cd7214edf85a61e4056d37/docs/syntax.md#module-declaration
> Module declaration takes optional self type parameter, which defines a constraint about a class when the module is mixed.



How about using it?

Reply:
Thanks! It was difficult to write the expression.
I tried it. Please check it. e79ad94412f8fe27a5d2c9a236657f14fc3f1e79